cmd/version.go

```diff
@@ -24,16 +25,70 @@ func versionString() string {
 	return v
 }
 
-func getCmdVersion(_ *state.GlobalState) *cobra.Command {
-	// versionCmd represents the version command.
-	return &cobra.Command{
+type versionCmd struct {
+	gs     *state.GlobalState
+	isJSON bool
+}
+
+func (c *versionCmd) run(cmd *cobra.Command, _ []string) error {
+	if !c.isJSON {
+		root := cmd.Root()
+		root.SetArgs([]string{"--version"})
+		_ = root.Execute()
+		return nil
+	}
+
+	details := consts.VersionDetails()
+	if exts := ext.GetAll(); len(exts) > 0 {
+		type extInfo struct {
+			Module  string   `json:"module"`
+			Version string   `json:"version"`
+			Imports []string `json:"imports"`
+		}
+
+		ext := make(map[string]extInfo)
+		for _, e := range exts {
+			if v, ok := ext[e.Path]; ok {
+				v.Imports = append(v.Imports, e.Name)
+				ext[e.Path] = v
+				continue
+			}
+
+			ext[e.Path] = extInfo{
+				Module:  e.Path,
+				Version: e.Version,
+				Imports: []string{e.Name},
+			}
+		}
+
+		list := make([]extInfo, 0, len(ext))
+		for _, v := range ext {
+			list = append(list, v)
+		}
+
+		details["extensions"] = list
+	}
+
+	jsonDetails, err := json.Marshal(details)
+	if err != nil {
+		return fmt.Errorf("failed produce a JSON version details: %w", err)
+	}
+
+	_, err = fmt.Fprintln(c.gs.Stdout, string(jsonDetails))
+	return err
```

Comment:
```suggestion
  return json.NewEncoder(c.gs.Stdout).Encode(details)
```

Reply:
Even if I like the simplicity of the approach that you're suggesting, I can't share the same feeling about returning plane errors from `Encode`, so I'd like to keep a human friendly wrapping `failed produce a JSON version details: ` here.

Reply:
On the other side it's still can be applied, done in e2535adc1f30926d98c37f25801b185c19797645